docs/design/fabric-v2+/fpc-key-dist.puml

```diff
@@ -213,111 +181,135 @@ group key distribution
+            and the keys are now being distributed to a new enclave.
+        - similarly, we cannot check version, sequence, endorsement
+            policy fields of the chaincode definition.
+        - however, we can enforce that the recipient enclave (1) matches
+            with the current chaincode definition, and (2) can do useful work.
+    end note
+    ERCC1       -> TLCC_Enclave1++: can_endorse(Enclave2_VK)
     return true
-    ERCC1       -> ERCC1            :   check SIG_ECC1
-    ERCC1       -> ERCC1            :   put_state(key=<CC_name, CC_ver, ECC1_S_PK, "keydist_started">, value=<encrypted_keys_message, key_distribution_message>)
+    ERCC1       -> ERCC1            :   check Sig_Enclave1
+    ERCC1       -> ERCC1            :   check Enclave1 previously registered cckey_distribution_message.Chaincode_EK
+    ERCC1       -> ERCC1            :   extract CC_Id from Enclave1_Credentials
+    ERCC1       -> ERCC1            :   check cckey_distribution_message.Chaincode_EK and get_state(CC_Id."Chaincode_EK") match
+    ERCC1       -> ERCC1            :   store <encrypted_cckeys_message, cckey_distribution_message> on the ledger
+    note over ERCC1
+        e.g., put_state(key=<"keydist_started".Enclave2_VK>, value=<encrypted_cckeys_message, cckey_distribution_message>)
+    end note
     return /'Admin1 -> ERCC1'/
   end group
 
   ref over Admin1, ERCC2
-    start-key-distribution endorsement
+    start-cckey-distribution endorsement
   end ref
 
-  Admin1 -> Orderer: submit start-key-distribution transaction
+  Admin1 -> Orderer: submit start-cckey-distribution transaction
 end group
 
 group complete key distribution
-  Admin2        -> Admin2           :   cc_target_message <- <CC_name, CC_ver, CH_id, org_id>
-  Admin2        -> ECC_Manager2     ++: ConfirmCCKeyDist(cc_target_message)
-  ECC_Manager2  -> ECC2             ++: ConfirmCCKeyDist(cc_target_message)
-  ECC2          -> Enclave2         ++: ConfirmCCKeyDist(cc_target_message)
-  Enclave2      -> ERCC2            ++: GetKeyDistStart(key=<CC_name, CC_ver, ECC2_S_PK, "keydist_started">)
-  return keydist_start_message <- value=<encrypted_keys_message, key_distribution_message> /'Enclave2 -> ERCC2'/
-  Enclave2      -> TLCC_Enclave2    ++: check_committed_data(keydist_start_message)
+  Admin2        -> Admin2           :   fill CC_Parameters
+  Admin2        -> E_Manager2       ++: RegisterCCKeyDist(CC_Parameters)
+  E_Manager2    -> ECC2             ++: RegisterCCKeyDist(CC_Parameters)
+  ECC2          -> Enclave2         ++: RegisterCCKeyDist(CC_Parameters)
+  Enclave2      -> Enclave2         :   check CC_Parameters matches internal one
+  Enclave2      -> ERCC2            ++: GetKeyDistStart(Public_Keys.Enclave2_VK)
+  return cckeydist_start_message <- <encrypted_cckeys_message, cckey_distribution_message> /'Enclave2 -> ERCC2'/
+  Enclave2      -> TLCC_Enclave2    ++: check_committed_data(cckeydist_start_message)
```

Comment:
If all these `check_committed_data` verify only ercc state (as i think so), we might better rename the function to `check_committed_ercc_state` or alike. If not, we should mention the namespace as otherwise this is a bit hard to read and understand

Reply:
done